.circleci/config.yml

```diff
@@ -0,0 +1,101 @@
+      - restore_cache: { key: 'gradle-cache-{{ checksum "versions.props" }}-{{ checksum "build.gradle" }}' }
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - run:
+          name: Make remote docker appear as local docker
+          command: sudo ssh -f -o "ExitOnForwardFailure yes" -F $HOME/.ssh/config -4N -L /var/run/docker.sock:/var/run/docker.sock $(for port in 2376 $(seq 32768 32999); do echo "-L ${port}:127.0.0.1:${port}"; done) remote-docker
+      - run:
+          name: Allow all users to access local docker
+          command: sudo chmod a+rw /var/run/docker.sock
+      - run: unset DOCKER_CERT_PATH DOCKER_HOST DOCKER_TLS_VERIFY && ./gradlew --no-daemon --parallel --stacktrace  --continue --max-workers=8 --profile build
+      - save_cache:
+          key: 'gradle-wrapper-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}'
+          paths: [ ~/.gradle/wrapper ]
+      - save_cache:
+          key: 'gradle-cache-v2-{{ checksum "versions.props" }}-{{ checksum "build.gradle" }}'
+          paths: [ ~/.gradle/caches ]
+      - run:
+          command: mkdir -p ~/junit && find . -type f -regex ".*/build/.*TEST.*xml" -exec cp --parents {} ~/junit/ \;
+          when: always
+      - deploy:
+          command: |
+            # publishing snapshots to bintray does not work, so we only publish from tag builds (not develop)
+            if [[ "${CIRCLE_TAG}" =~ [0-9]+(\.[0-9]+)+(-[a-zA-Z]+[0-9]*)* ]]; then
+              ./gradlew --no-daemon --parallel --stacktrace  --continue --max-workers=8 --profile publish
+            else
+              ./gradlew --no-daemon --parallel --stacktrace  --continue --max-workers=8 --profile publishToMavenLocal
+              mkdir -p $CIRCLE_ARTIFACTS/poms
+              find . -name 'pom-default.xml' -exec cp --parents {} $CIRCLE_ARTIFACTS/poms \;
+            fi
+      - store_test_results: { path: ~/junit }
+      - store_artifacts: { path: ~/artifacts }
+
+  build-11:
+    docker: [{ image: 'circleci/openjdk:11-node' }]
+    resource_class: xlarge
+    environment:
+      GRADLE_OPTS: -Dorg.gradle.console=plain -Dorg.gradle.internal.launcher.welcomeMessageEnabled=false
+      CIRCLE_TEST_REPORTS: /home/circleci/junit
+      CIRCLE_ARTIFACTS: /home/circleci/artifacts
+      _JAVA_OPTIONS: '-XX:ActiveProcessorCount=8 -Xmx2048m'
+    steps:
+      - checkout
+      - restore_cache: { key: 'gradle-wrapper-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}' }
+      - restore_cache: { key: 'gradle-cache-{{ checksum "versions.props" }}-{{ checksum "build.gradle" }}' }
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - run:
+          name: Make remote docker appear as local docker
+          command: sudo ssh -f -o "ExitOnForwardFailure yes" -F $HOME/.ssh/config -4N -L /var/run/docker.sock:/var/run/docker.sock $(for port in 2376 $(seq 32768 32999); do echo "-L ${port}:127.0.0.1:${port}"; done) remote-docker
```

Comment:
Personally, I'd recommend using the `machine` executor instead of wiring up lots of funky SSH stuff.

Reply:
Any idea how to get Java 11 working on the `machine` executor?

Reply:
You can just download openjdk11 and untar it.

Reply:
I'm going to merge as-is for now to get Java 11 support in and probably update the java-library-oss template in the future at some point instead.

Reply:
nvm, I found a simple example to copy so I've switched it to machine executors now.